### PR TITLE
refactor: unit test

### PR DESCRIPTION
**Pull request by juanpicado**
**Type:** refactor

The following has been addressed in the PR:

*  There is a related issue? No

**Description:**
* Unit test currently runs against npmjs (that's bad)
* Mock with verdaccio registry and goes 100% offline
* Remove all Jest warnings
* Stop ussage of `assert` node build-it method in favor of Jest mathers
* Improve maintenance

(This do not affect UI unit test, those will be treated in a different ticket)

### CODE REVIEW

File: jest.config.unit.js [hunk 1]
```diff
@@ -4,21 +4,41 @@ module.exports = {
...
-    '__snapshots__'
+    '__snapshots__',
+    '<rootDir>/build',
+  ],
+  coveragePathIgnorePatterns: [
+    'node_modules',
+    'fixtures',
+    '<rootDir>/test',
   ],
+  // moduleFileExtensions: ["js"],
```
Comment (ayusharma) [inline jest.config.unit.js]: Do we need this ? Already commented.

File: src/lib/constants.js [hunk 2]
```diff
@@ -1,9 +1,13 @@
...
 
+export const DEFAULT_PORT = '4873';
+export const DEFAULT_DOMAIN = 'localhost';
+
 export const HEADERS = {
   JSON: 'application/json',
   JSON_CHARSET: 'application/json; charset=utf-8',
-  OCTET_STREAM: 'application/octet-stream',
+  OCTET_STREAM: 'application/octet-stream; charset=utf-8',
+  TEXT_CHARSET: 'text/plain; charset=utf-8',
```
Comment (ayusharma) [inline src/lib/constants.js]: we can have a constant for `text/html` as well. 

In UI:
screen shot 2018-06-28 at 21 44 14

File: src/lib/constants.js [hunk 3]
```diff
@@ -57,6 +74,12 @@ export const API_ERROR = {
   NOT_FILE_UPLINK: 'file doesn\'t exist on uplink',
```
Comment (ayusharma) [inline src/lib/constants.js]: We should ignore the short forms. It should be `file does not exist on uplink`

File: src/lib/constants.js [hunk 4]
```diff
@@ -57,6 +74,12 @@ export const API_ERROR = {
...
   MAX_USERS_REACHED: 'maximum amount of users reached',
   VERSION_NOT_EXIST: 'this version doesn\'t exist',
```
Comment (ayusharma) [inline src/lib/constants.js]: Here also, `this version does not exist`

File: test/unit/partials/mock-store/corrupted-package/package.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+	"name": "corrupted-package"
```
Comment (ayusharma) [inline test/unit/partials/mock-store/corrupted-package/package.json]: comma `,` missed.

File: src/lib/constants.js (hunk 2 above)
Comment (juanpicado) [reply src/lib/constants.js]: I think all `text/html` should be utf-8. What do you think?

File: src/lib/constants.js (hunk 3 above)
Comment (juanpicado) [reply src/lib/constants.js]: the problem is some plugins still use them. I'll remove that when I manage to normalize such plugins.

File: src/lib/constants.js (hunk 4 above)
Comment (juanpicado) [reply src/lib/constants.js]: Same problem above.

File: test/unit/partials/mock-store/corrupted-package/package.json (hunk 5 above)
Comment (juanpicado) [reply test/unit/partials/mock-store/corrupted-package/package.json]: 😝 ... it's a corrupted package, that's the point hehe ..

Comment (ayusharma) [reply test/unit/partials/mock-store/corrupted-package/package.json]: shit.. I totally missed the name =D

File: src/lib/constants.js (hunk 2 above)
Comment (ayusharma) [reply src/lib/constants.js]: Not sure about this.

**Pull request merged**